### DISCUSSION TITLE
Alpine V3 stale state with data

### DISCUSSION THREAD

**Discussion by Gazler** [livewire/livewire#3505] [7. Bugs (archived)] [answered]
Hi,

First of all, if this is the wrong place for this issue let me know and I can move it to the relevant location.

When integrating AlpineJS with a Phoenix LiveView application, I encountered an issue where `x-text` from AlpineJS was not being updated when `x-data` was updated in response to an event over the WebSocket. You can see in the video below that hte update does not occur when the event triggers. This was not the case with Alpine V2.

It looks like potentially this event that triggers after the virtual DOM update should trigger effects, however it has no effect on the `x-text`.

`<div x-data="{counter: '{{ $counter }}'}" x-text="counter"></div>`

https://github.com/livewire/livewire/blob/5cffdbfe4204e0d932cde6f8c9bdf63f0928b4d4/js/component/SupportAlpine.js#L16

I was also able to replicate this issue in LiveWire with the following component:

```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Counter extends Component
{
    public $counter = 0;

    public function increament(){
        $this->counter+=1;
    }

    public function decreament(){
        $this->counter-=1;
    }

    public function render()
    {
        return view('livewire.counter');
    }
}

```

```html
counter.blade.php
<div>
    <div x-data="{counter: '{{ $counter }}'}" x-text="counter"></div>
    <button type="button" wire:click="increament" >Increament</button>
    <input type="number" wire:model="counter" value="" readonly >
    <button type="button" wire:click="decreament" >Decrement</button>
</div>
```

I've never used LiveWire or Laravel before, so I hope this example is representative of what is going on.

https://user-images.githubusercontent.com/477511/129023454-fe6d2dc3-0742-4ff6-87be-51c0b37d0cb7.mp4

**Comment by Gazler** [comment, marked as answer]
Looks like this is a duplicate of https://github.com/livewire/livewire/discussions/3284 - I was looking in the issues section and not the discussions when I searched for this issue!

**Reply by joshhanley** [reply]
@Gazler glad you found that thread 🙂 Did you manage to solve the issue with any of the information there?

If not, let me know, and we can have a look to see if we can work it out!

**Reply by Gazler** [reply]
@joshhanley Thanks for the reply. Adding a unique id to the element that is changed from the server did the trick since the entire element is replaced in that case, causing `x-data` to run again.

**Reply by joshhanley** [reply]
@Gazler Nice! no worries 🙂
